src/coreComponents/dataRepository/Group.cpp

```diff
@@ -267,6 +267,42 @@ string Group::dumpInputOptions() const
   return rval;
 }
 
+string Group::dumpChildrenName() const
+{
+  if( getSubGroups().size() > 0 )
+  {
+    string str = "";
+    forSubGroupsIndex( [&] ( localIndex i, const Group & subGroup )
+    {
+      str += (i==0 ? subGroup.getName() : ", "+subGroup.getName());
+    } );
+    return str;
+  }
+  else
+  {
+    return "No children";
+  }
```

Comment:
Why don't you simply do
```c++
std::vector< string > children;
forSubGroups( [&]( Group const & subGroup ){ children.push_back( subGroup.getName() ); } );
return "{" + stringutilities::join( children, ", " ) + "}";
```
And I would make no difference for empty lists: it's empty, it's empty.

Reply:
You're right, the methods `dumpChildrenNames()` and `dumpChildrenNames()` should return nothing when no children are found.
But the user feedback is that they prefer to have a specific message when no children is found. I will output a specific message in that case.

Reply:
You're right, the `dumpChildrenNames()` method should return nothing when there are no children. However, the user feedback is that they prefer to have a more verbose message, so I'm making a distinct message for that case (and doing that for dumpWrappersNames() too).